docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+# Contributing to Devworkspace-Operator
+
+Hello there! Thank you for choosing to the contributing to devfile/devworkspace-operator. Navigate through the following to understand more about contributing here.
+
+- [Contributing to Devworkspace-Operator](#contributing-to-devworkspace-operator)
+- [How to Contribute](#how-to-contribute)
+  - [Set up your Development Environment](#set-up-your-development-environment)
+      - [Running the controller locally.](#running-the-controller-locally)
+  - [Testing Changes](#testing-changes)
+      - [Test run the controller](#test-run-the-controller)
+      - [Developing Webhooks](#developing-webhooks)
+  - [Signing-off on Commits](#signing-off-on-commits)
+
+# How to Contribute
+To contribute to the devworkspace-operator project, developers should follow the fork and pull request workflow. 
+## Set up your Development Environment
+
+
+**1.** Fork [Devworkspace operator](https://github.com/devfile/devworkspace-operator) repository.
+
+**2.** Clone your forked copy of the project.
+
+```
+git clone https://github.com/<your-github-username>/devworkspace-operator.git
+```
+
+#### Running the controller locally.
+
+In the steps listed below, we set up the development environment using a minikube cluster. 
+
+**1.** Start the minikube cluster.
+
+```
+minikube start
+```
+
+**2.** Enable the ingress add-on for your minikube cluster.
+
+```
+minikube addons enable ingress
+```
+
+**3.** Set the namespace environment variable for the development environment to avoid changes inside the default namespace.
+
+```
+export NAMESPACE="devworkspace-controller"
+```
+
+**4.** Install the kubernetes certificate management controller to generate and manage TLS certificates for your cluster. 
+
+```
+make install_cert_manager.
+```
+Please note that the above step is not specific to minikube. The cert-manager is required for all deployments on kubernetes.
+
+
+**5.** Install the dependencies for running the devworkspace-operator in your cluster.
+
+```
+make install
+```
+
+**6.** Scale down the replicas controller-manager pods to 0.
+
+```
+kubectl patch deployment/devworkspace-controller-manager --patch "{\"spec\":{\"replicas\":0}}" -n $NAMESPACE
+```
+
+**7.** Run the devworkspace-operator.
+
+```
+make run
+```
+
+This will run the devworkspace-controller in your local system.
+
+**8.** Make your changes in the new branch and test the changes.
```

Comment:
Since there was no previous mention of creating a feature branch, it might be worth to change this to:
"Make your changes in _a_ new branch and test the changes."